codelabs/codelab-03/README.md

```diff
@@ -0,0 +1,143 @@
+
+For comparison, try some other URLs in the script. If you use the link for the index file from codelab-02, the time should be on the order of .5 seconds (recall we used us-east-1 which is located in northern Virginia).
+
+#### CloudFront
+
+Lets return to using the [AWS GUI](https://console.aws.amazon.com/console/home?region=us-east-1#), this time navigating to CloudFront, which can be found under *Storage & Content Delivery*.
+
+From the S3 CloudFront, click `Create Distribution`. This will allow us to tell AWS which origin to use for our content and configure settings. There are two types of delivery CloudFront can use, *web* and *RTMP*.
+
+Click the top `Get Started` button under the _Web_ heading. This is for static (like our canyon image) and dynamic content. _RTMP_ is best for streaming.
+
+Now, click into the *Origin Domain Name* text box. You will see a list of possible origins for content. Select the bucket created in the first part of this tutorial. We'll be using the default values for all other fields in this tutorial.
+
+Scroll to the bottom and click the blue `Create Distribution` in the bottom-left. You will then be on a page showing your distributions that will look something like this.
+
+![dist_status_in_progress Page](../../../media/codelabs/codelab-03/dist_status_in_progress.png)
+
+Note that the status is *in progress*, deployment may take several minutes. When complete, the status will change to *deployed*. Then, go ahead and grab the _Domain Name_ (it should end in .cloudfront.net).
+
+Now we're ready to test. The new URL of our content will be of the form "http://domainName/objectName". So it should look something like `http://3x4mpl3.cloudfront.net/canyon.jpg`. Try this in your browser.
+
+Edit the shell script again and run it a few times. Does the total time change?
+
+![lat_test_on_canyon_in_cloudfront_sydney Page](../../../media/codelabs/codelab-03/lat_test_on_canyon_in_cloudfront_sydney.png)
+
+Now suppose we want to change the image we're serving. Using the [AWS GUI](https://console.aws.amazon.com/console/home?region=us-east-1#) again, delete the current `canyon.jpg` and replace it with [this one](https://s3.amazonaws.com/cmsc389l-ialock-lab3/canyon.jpg). *Note: the names are identical.* 
+
+Verify you can reach it by the S3 link from your browser. Then try to reach it with your CloudFront link... and you might not be able to. This is because when CloudFront populates its Edge Cache, each object is given a expiration date called Time-To-Live (24 hours by default). When you used your CloudFront link, the Edge server gave you the object it had because it was still "fresh".
+
+### Assignment
+
+Your assignment for this codelab is to reference the [documentation](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AddRemoveReplaceObjects.html) and issue and invalidation via the AWS CLI.
+
+### Submission
+
+You will be submitting three screenshots 
```

Comment:
this is a bit hard to parse. I'd recommend a numbered list?

1. Screenshot of `lat_test.sh` output on `canyon.jpg` in Sydney S3
2. Two other screenshots of `lat_test.sh` output on two different images from the internet.
3. Screenshot of `lat_test.sh` output on `canyon.jpg` in CloudFront
4. Screenshot showing the invalidation command you used and its console output
5. 3-5 sentences explaining the tradeoff associated with time duration of objects in the cache.